substrate/client/service/src/builder.rs

```diff
@@ -664,6 +666,22 @@ where
 	)
 	.into_rpc();
 
+	// Part of the RPC v2 spec.
+	// An archive node that can respond to the `archive` RPC-v2 queries is a node with:
+	// - state pruning in archive mode: The storage of blocks is kept around
+	// - block pruning in archive mode: The block's body is kept around
+	let is_archive_node = config.state_pruning.as_ref().map(|sp| sp.is_archive()).unwrap_or(false) &&
+		config.blocks_pruning.is_archive();
+	let genesis_hash = client.hash(Zero::zero()).ok().flatten().expect("Genesis block exists; qed");
```

Comment:
Are things too generic to use `0` here instead of `Zero::zero()`?

Reply:
Yep, the `.hash` needs a ` NumberFor<Block>`; this is similar [here](https://github.com/paritytech/polkadot-sdk/blob/lexnv/enable-archive/substrate/client/service/src/builder.rs#L931)

Reply:
nit: just create this if is_archive_node == true?